docs/desired-state-feedback-specification.md

```diff
@@ -0,0 +1,136 @@
+
+The feedback is comprised by overall information for the current status of the ongoing update process, information for the status for each of the identified components and optional unstructured message. This message is meant to support projects with the need to transfer custom data such as a human readable progress information or internally used status codes.
+Optionally, the feedback can be defined at baseline level by specifying additional information for the baseline the feedback is associated with. If not present, the feedback is considered to be related at domain level.
+
+### Desired State Feedback Data Model
+
+The following table describes all supported properties and sections of the Desired State specification:
+
+| Property | Type | Description |
+| - | - | - |
+| **General properties** | | |
+| status | string | Status of the Desired State Feedback. The supported statuses are listed below |
+| baseline | string | Title of the baseline, the Desired State Feedback is associated with. Optional |
+| message | string | Info message for the Desired State Feedback |
+| actions | JSON array | Set of Desired State Feedback actions |
+| **Action properties** | | |
+| component | JSON object | The component, which is linked with the action |
+| status | string | Status the of action. The supported statuses are listed below |
+| progress | int | Progress of the action in percentage |
+| message | string | Info message for the action |
+| **Config properties** | | |
+| key | string | Key of the configuration property |
+| value | string | Value of the configuration property |
+| **Component properties** | | |
+| id | string | Identifier of the component |
+| version | string | Version of the component |
+| config | JSON object | Set of component-specific configuration properties as key/value pairs |
+
+### Supported Desired State Feedback statuses
+
+The list of the supported feedback statuses :
+
+| Status | Description |
+| - | - |
+| IDENTIFYING | Denotes that action identification process has started |
+| IDENTIFIED | Denotes that required actions are identified |
+| IDENTIFICATION_FAILED | Denotes some or all of the required actions failed to be indentified |
+| RUNNING | Denotes that identified actions are currently executing |
+| DOWNLOADED_COMPLETED | Denotes that identified actions are downloaded successfully |
+| DOWNLOADED_FAILED | Denotes that not all of the identified actions are downloaded successfully |
+| UPDATE_COMPLETED | Denotes that identified actions are updated successfully |
+| UPDATE_FAILED | Denotes that not all identified actions are updated successfully |
+| ACTIVATION_COMPLETED | Denotes that identified actions are activated successfully |
+| ACTIVATION_FAILED | Denotes that not all identified actions are activated successfully |
+| COMPLETED | Denotes that identified actions completed successfully |
+| INCOMPLETE | Denotes that not all of the identified actions completed successfully |

```

Comment:
Would make `COMPLETED` and `INCOMPLETE` consistent, so either `COMPLETED` and `INCLOMETED` or my preferred option `COMPLETE` and `INCOMPLETE`. In a similar fashion one could change the states which end with `_COMPLETED` to `_COMPLETE` .

Reply:
Please, check my answer above for the BfB message definitions and value constant.